js/views/conversation_search_view.js

```diff
@@ -97,12 +97,19 @@
         createConversation: function() {
             var conversation = this.new_contact_view.model;
             if (this.new_contact_view.model.isValid()) {
-                ConversationController.findOrCreatePrivateById(
-                    this.new_contact_view.model.id
-                ).then(function(conversation) {
-                    this.trigger('open', conversation);
-                    this.initNewContact();
-                    this.resetTypeahead();
+                var accountManager = window.getAccountManager();
+                accountManager.server.getKeysForNumber(conversation.id)
+                .then(function(response) {
```

Comment:
First, I think it would be a good idea to extract the `findOrCreatePrivateById()` call into a different method and then call it here. That would make this function a whole lot simpler.

Second, the `.then` at the same indentation level of the `getKeysForNumber()` call makes it a little tough to follow.  Something like this would help:

```javascript
accountManager.server
  .getKeysForNumber(conversation.id)
  .then(function(response) {
    // stuff
  }.bind(this), function(error) {
    // stuff
  })
```